src/Inventory.zig

```diff
@@ -1555,9 +1556,80 @@ pub const Command = struct { // MARK: Command
 	const UpdateBlock = struct { // MARK: UpdateBlock
 		source: InventoryAndSlot,
 		pos: Vec3i,
+		drop: BlockDrop,
 		oldBlock: Block,
 		newBlock: Block,
 
+		const BlockDrop = struct {
+			dir: Neighbor,
+			min: Vec3f,
+			max: Vec3f,
+
+			pub fn drop(self: BlockDrop, pos: Vec3i, newBlock: Block, _drop: main.blocks.BlockDrop) void {
+				if(newBlock.solid() and newBlock.collide()) {
+					self.dropOutside(pos, _drop);
+				} else {
+					self.dropInside(pos, _drop);
+				}
+			}
+			fn dropInside(self: BlockDrop, pos: Vec3i, _drop: main.blocks.BlockDrop) void {
+				for(_drop.items) |itemStack| {
+					main.server.world.?.drop(itemStack.clone(), self.insidePos(pos), self.dropDir(), self.dropVelocity());
+				}
+			}
+			fn dropOutside(self: BlockDrop, pos: Vec3i, _drop: main.blocks.BlockDrop) void {
+				for(_drop.items) |itemStack| {
+					main.server.world.?.drop(itemStack.clone(), self.outsidePos(pos), self.dropDir(), self.dropVelocity());
+				}
+			}
+			fn insidePos(self: BlockDrop, _pos: Vec3i) Vec3d {
+				const pos: Vec3d = @floatFromInt(_pos);
+				return pos + self.randomOffset();
+			}
+			fn randomOffset(self: BlockDrop) Vec3d {
+				const itemMargin = itemHitBoxMargin();
+				const max: Vec3d = @min(@as(Vec3d, @splat(1.0)) - itemMargin, @max(itemMargin, self.max - itemMargin));
+				const min: Vec3d = @min(max, @max(itemMargin, self.min + itemMargin));
+				const center: Vec3d = (max + min)/@as(Vec3d, @splat(2));
+				const width: Vec3d = (max - min)/@as(Vec3d, @splat(2));
+				const factors = main.random.nextDoubleVectorSigned(3, &main.seed)*@as(Vec3d, @splat(0.5));
+				const offset = center + width*factors;
+				std.log.debug("min {d:.2} max {d:.2} offset {d:.2}", .{min, max, offset});
```

Comment:
This seems like it would needlessly spam the log, I assume you only used this for debugging?

Reply:
```suggestion
```

Yeah, it can be removed.